app/controllers/api/template_versions_controller.rb

```diff
@@ -240,11 +240,33 @@ def set_jurisdiction_template_version_customization
   end
 
   def jurisdiction_template_version_customization_params
-    params.require(:jurisdiction_template_version_customization).permit(
-      customizations: {
-        requirement_block_changes: {
+    # This requires a multi step permit structure in order for this to work correctly
+
+    # Step 1: Permit the outer structure without the dynamic keys
+    permitted_params =
+      params.require(:jurisdiction_template_version_customization).permit(
+        customizations: {
+          requirement_block_changes: {
+          }
         }
-      }
-    )
+      )
+
+    # Step 2: Iterate through each dynamic key and permit nested attributes
+    if permitted_params[:customizations][:requirement_block_changes].present?
+      permitted_params[:customizations][
+        :requirement_block_changes
+      ].each do |key, value|
+        permitted_params[:customizations][:requirement_block_changes][
+          key
+        ] = value.permit(
+          :tip,
+          enabled_elective_field_ids: [],
+          enabled_elective_field_reasons: {
+          }
+        )
+      end
+    end
+
+    permitted_params
```

Comment:
seems like there should be a better way to do this. Do you have an example request payload looks like? Could it be changed to pass an array of changes instead of an object? e.g. `[{key: key, tip: tip, enabled_elective_field_ids: [], enabled_elective_field_reasons: {} }]`

Reply:
I have exhausted every other avenue on this. Spent half of the morning on it. If you dont want to approve this, we can leave it as-is because the dependabot message is a nothingburger anyway.

Please see these chatgpt conversation:

https://chatgpt.com/share/6737d64f-d5fc-800d-9ab4-dcb85d1be70d

https://chatgpt.com/share/6737d664-4038-800d-959f-dc69a4f8b7c6